ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryCleanupService.java

```diff
@@ -99,13 +99,15 @@ public static class QueryCleanupTask implements Runnable {
     private final Optional<String> topologyName;
     private final String pathName;
     private final boolean isTransient;
+    private final boolean isPersistent;
     private final ServiceContext serviceContext;
 
     public QueryCleanupTask(
         final ServiceContext serviceContext,
         final String appId,
         final Optional<String> queryId,
         final boolean isTransient,
+        final boolean isPersistent,
```

Comment:
a nice cleanup, in line with the other comments I made, would be to name these booleans with what it is that they do instead of the property of what's passing them in (that breaks the abstraction barrier - and yes, I know I originally wrote that 😆)

for example, flags like `shouldHardDeleteSchemas` or `shouldHardDeleteInternalSchemas` make a lot more sense in the context of this class. Alternatively, we can refactor this class into multiple different cleanup tasks that each do only one thing and then register them based on what we need to do.

Reply:
> a nice cleanup, in line with the other comments I made, would be to name these booleans with what it is that they do instead of the property of what's passing them in (that breaks the abstraction barrier - and yes, I know I originally wrote that 😆)
> 
> for example, flags like `shouldHardDeleteSchemas` or `shouldHardDeleteInternalSchemas` make a lot more sense in the context of this class. Alternatively, we can refactor this class into multiple different cleanup tasks that each do only one thing and then register them based on what we need to do.

I prefer to keep the name isTransient, because of the following part:
```
 if (isTransient && queryId.isPresent()) {
        throw new IllegalArgumentException("Transient Queries can not have named topologies");
 }
```